plugins/Marketplace/Consumer.php

```diff
@@ -65,4 +70,18 @@ public function isValidConsumer()
 
         return $this->isValid;
     }
+
+    public function getConsumerPluginLicenseStatus(): array
+    {
+        if (!$this->pluginLicenseStatus) {
+            $consumer = $this->getConsumer();
+            if (!empty($consumer['licenses'])) {
+                foreach ($consumer['licenses'] as $license) {
+                    $this->pluginLicenseStatus[$license['plugin']['name']] = $license['status'];
+                }
+            }
+        }
+
+        return $this->pluginLicenseStatus ?? [];
```

Comment:
I guess this is meant as kind of caching the result. But I wonder what it actually does if a consumer does not have a license at all. It would then still call `getConsumer` on every call, as `pluginLicenseStatus` is never filled. Also directly setting an array key of a property, which might be `null`, would most likely throw a notice. To properly cache any result I guess something like this would be better:

```suggestion
        if (null === $this->pluginLicenseStatus) {
            $consumer = $this->getConsumer();
            $this->pluginLicenseStatus = [];
            if (!empty($consumer['licenses'])) {
                foreach ($consumer['licenses'] as $license) {
                    $this->pluginLicenseStatus[$license['plugin']['name']] = $license['status'];
                }
            }
        }

        return $this->pluginLicenseStatus;
```

Reply:
makes sense :+1: